builtin/logical/database/backend.go

```diff
@@ -332,14 +345,17 @@ func (b *databaseBackend) GetConnectionWithConfig(ctx context.Context, name stri
 		name:                 name,
 		runningPluginVersion: pluginVersion,
 	}
-	oldConn := b.connections.Put(name, dbi)
-	if oldConn != nil {
-		err := oldConn.Close()
+	conn, ok := b.connections.PutIfEmpty(name, dbi)
+	if !ok {
+		// this is a bug
+		b.Logger().Error("Error: there was a race condition adding to the database connection map")
```

Comment:
Should we mention that we hit a bug here? A user could then report it.

Should this be `Warn()` like line 355 below?